client/client.go

```diff
@@ -488,6 +490,73 @@ func (c *client) GetRegion(ctx context.Context, key []byte) (*Region, error) {
 	return c.parseRegionResponse(resp), nil
 }
 
+func (c *client) waitForRegion(ctx context.Context, key []byte) (*pdpb.GetRegionResponse, error) {
+	childContext, cancel := context.WithCancel(ctx)
+
+	ch := make(chan *pdpb.GetRegionResponse, len(c.connMu.clientConns))
+
+	c.connMu.RLock()
+	for _, conn := range c.connMu.clientConns {
```

Comment:
if the leader does not change, the connection with another member will not exist.